src/Mono.Android/Android.App/Activity.cs

```diff
@@ -12,6 +12,18 @@ partial class Activity {
 			return this.FindViewById (id)!.JavaCast<T> ();
 		}
 
+		// See: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/View.java;l=25322
+		public T RequireViewById<T> (int id)
+			where T : Android.Views.View
+		{
+			var view = FindViewById<T> (id);
+			if (view == null)
+			{
+				throw new ArgumentException ($"ID 0x{id:X} does not reference a View of type '{typeof (T)}' inside this View");
```

Comment:
`ArgumentException` should also specify the parameter name, `id`.

Reply:
I changed it to `Java.Lang.IllegalArgumentException` to match the Java/non-generic version. That one doesn't have a second argument, but still seems OK.